scripts/tb.sh

```diff
@@ -0,0 +1,290 @@
+            ;;
+        menuconfig)
+            RECIPE_ARG="linux-tb"
+            ;;
+        *)
+            usage_error "Wrong positional argument"
+            ;;
+    esac
+
+    case $RECIPE_ARG in
+        skl|grub|grub-efi|linux-tb|tb-minimal-image)
+            ;;
+        *)
+            usage_error "Wrong recipe"
+            ;;
+    esac
+}
+
+menuconfig() {
+    if [ -d "build/workspace/sources/$RECIPE_ARG" ]; then
+        kas-container shell "$KAS_YAML" \
+            -c "devtool menuconfig $RECIPE_ARG"
+    else
+        error_msg "First use ./tb.sh modify linux-tb"
+    fi
+}
+
+modify_recipe() {
+    if [ ! -d "build/workspace/sources/$RECIPE_ARG" ]; then
+        kas-container shell "$KAS_YAML" \
+            -c "devtool modify $RECIPE_ARG"
+    else
+        error_msg "Source is already fetched"
+    fi
+}
+
+reset_recipe() {
+    if [ -d "build/workspace/sources/$RECIPE_ARG" ]; then
+        kas-container shell "$KAS_YAML" \
+            -c "devtool reset $RECIPE_ARG"
+    else
+        error_msg "Nothing to reset"
+    fi
+}
+
+build_recipe() {
+    if [ "$RECIPE_ARG" == "tb-minimal-image" ]; then
+        rm -rf "build/workspace/sources/grub/gnulib" 2&>/dev/null|| true
+        rm -rf "build/workspace/sources/grub-efi/gnulib" 2&>/dev/null || true
+        kas-container shell "$KAS_YAML" \
+            -c "devtool build-image $RECIPE_ARG"
+        return
+    fi
+    if [ ! -d "build/workspace/sources/$RECIPE_ARG" ]; then
+        modify_recipe
+    elif [ "$RECIPE_ARG" == "grub" ] || [ "$RECIPE_ARG" == "grub-efi" ]; then
+        rm -rf "build/workspace/sources/$RECIPE_ARG/gnulib"
+    fi
+
+    kas-container shell "$KAS_YAML" \
+        -c "devtool build $RECIPE_ARG"
+}
+
+deploy_recipe() {
+    local work_dir="build/tmp/work"
+    local deploy_dir="build/tmp/deploy/images/genericx86-64"
+    local core_path="$work_dir/core2-64-tb-linux"
+    local genericx86_path="$work_dir/genericx86_64-tb-linux"
+    local kernel_path="$genericx86_path/linux-tb/6.6.1"
+    local partition_path_cmd=
+    local device_path_cmd=
+    local partition_path=
+    local device_path=
+    local tmp_dir=
+    local grub_install_cmd=
+    local remote=
+    local remote_path=
+
+    case $RECIPE_ARG in
+        skl)
+            sudo rsync -chavP "$core_path/skl/git/image/" "$DESTINATION_ARG"
+            sudo rsync -chrtvP --inplace "$core_path/skl/git/deploy-skl/skl.bin" "$DESTINATION_ARG/boot"
+            ;;
+        grub)
+            remote="${DESTINATION_ARG%:*}"
+            remote_path="${DESTINATION_ARG#*:}"
+            # shellcheck disable=SC2016
+            partition_path_cmd="df \"$remote_path/boot\" | tail -1 |  awk '{print "'$1}'\'
+            # shellcheck disable=SC2016
+            device_path_cmd='echo -n /dev/"$(lsblk -ndo pkname "$('$partition_path_cmd')")"'
+            case $DESTINATION_ARG in
+                *:*)
+                    rsync -chavP --exclude "boot" "$core_path/grub/2.06/image/" "$DESTINATION_ARG"
```

Comment:
Same here, first task in next phase is upgrading GRUB to 2.12.

Reply:
[3ba2257](https://github.com/3mdeb/meta-trenchboot/pull/29/commits/3ba2257a93bfbfc634d83f4f3c4f5bd194fc63c0)